### PR TITLE
Merge NestedTeamsPreview branch to master

### PR DESCRIPTION
**Pull request by gmlewis**

### CODE REVIEW

**Comment by gmlewis** [comment]
All CLAs signed. Changing label.

**Comment by gmlewis** [comment]
libraryVersion needs bumping.

**Comment by elliotbeach** [comment]
@gmlewis Would you like me to do that in another PR? I can’t here. I realize now that I could have opened this PR myself.

**Comment by gmlewis** [comment]
Thanks @elliotbeach, but I think we are good now.  This has been a fun little exercise.
I'm going to ask that @shurcooL review it after I look it over one time, and he can merge if I haven't made any mistakes.

File: github/orgs_teams.go [hunk 1]
```diff
@@ -107,16 +111,50 @@ func (s *OrganizationsService) GetTeam(ctx context.Context, team int) (*Team, *R
 	return t, resp, nil
 }
 
+// NewTeam represents a team to be created or modified.
+type NewTeam struct {
+	Name         string   `json:"name"` // Name of the team. (Required.)
+	Description  *string  `json:"description,omitempty"`
+	Maintainers  []string `json:"maintainers,omitempty"`
+	RepoNames    []string `json:"repo_names,omitempty"`
+	ParentTeamID *string  `json:"parent_team_id,omitempty"`
```
[the remark this answers is not in the document]
Comment (alindeman) [reply github/orgs_teams.go]: To remove a team from the hierarchy and place it at the top level, GitHub allows the `PATCH` verb to be used with the body `{"parent_team_id": null}`. Is it possible with `omitempty` to send this kind of request right now?

But on the other hand, if we remove `omitempty` naively, we force `ParentTeamID` to be specified any time we edit the team, even if we only want to edit other attributes.

How do we balance both use cases? Is there prior art?

[the remark this answers is not in the document]
Comment (elliott-beach) [reply github/orgs_teams.go]: It's not possible to specify the id as null with `omitempty` - similar issues have been encountered in #181 and #236.

Could we add another function, MakeTeamTopLevel, which sets `{"parent_team_id": null}`?

**Comment by shurcooL** [comment]
This is waiting on GitHub indirectly via #748.

File: github/orgs_teams.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (elliott-beach) [reply github/orgs_teams.go]: @shurcooL Could we get your opinion on this point?

[the remark this answers is not in the document]
Comment (shurcooL) [reply github/orgs_teams.go]: Definitely similar to #236, thanks for referencing that @elliott-beach.

I suspect we'll need a separate method or something to force sending the `null` via another internal struct.

**Comment by elliott-beach** [comment]
I will make a pull request to add a method for setting `parent_team_id` to null, but I don't see a reason this couldn't be merged.

**Comment by shurcooL** [comment]
We'll need to resolve the conflict with `libraryVersion` bump (it should become 14 now). I'll do that by rebasing this branch on latest `master`. Maybe it's best to bump it just before merging the PR, to avoid the possibility of other conflicts. So I'll leave out 98b35b2be8ae24fcea8e1f84daebb0f59260d593 for now.

I notice that d51e256270584c5fb21a8ab2c9795fcf2d7de652 commit message says "Fixes #714.", but I don't see the new endpoint https://developer.github.com/v3/orgs/teams/#list-child-teams implemented yet. Doesn't that mean #714 is not going to be fully resolved yet?

**Comment by gmlewis** [comment]
I'm sorry, that Fix message was my fault... I thought it was resolved, but apparently not.

**Comment by elliott-beach** [comment]
It’s more my fault - I will fix tomorrow.

**Comment by gmlewis** [comment]
@elliott-beach - what is the status here - as far as I can tell, I can squash and merge this, correct?

**Comment by elliott-beach** [comment]
Now that the `ListChildTeams` endpoint is added, this should be good.

**Comment by gmlewis** [comment]
OK, thank you, @shurcooL!  I will squash and merge.

**Pull request merged**